apps/server/src/modules/files-storage/controller/files-storage.controller.ts

```diff
@@ -108,32 +110,60 @@ export class FilesStorageController {
 		const bytesRange = req.header('Range');
 
 		// Call download method with either defined or undefined bytes range.
-		const res = await this.filesStorageUC.download(currentUser.userId, params, bytesRange);
+		const fileResponse = await this.filesStorageUC.download(currentUser.userId, params, bytesRange);
 
 		// Destroy the stream after it has been closed.
-		req.on('close', () => res.data.destroy());
-
-		// If bytes range has been defined, set Accept-Ranges and Content-Range HTTP headers
-		// in a response and also set 206 Partial Content HTTP status code to inform the caller
-		// about the partial data stream. Otherwise, just set a 200 OK HTTP status code.
-		if (bytesRange) {
-			response.set({
-				'Accept-Ranges': 'bytes',
-				'Content-Range': res.contentRange,
-			});
-
-			response.status(HttpStatus.PARTIAL_CONTENT);
-		} else {
-			response.status(HttpStatus.OK);
-		}
+		req.on('close', () => fileResponse.data.destroy());
+
+		// Set headers and status.
+		setBytesRangeHeader(response, fileResponse, bytesRange);
+
+		// Return StreamableFile with stream data and options that will additionally set
+		// Content-Type, Content-Disposition and Content-Length headers in a response.
+		const streamableFile = FilesStorageMapper.mapToStreamableFile(fileResponse);
+
+		return streamableFile;
+	}
+
+	@ApiOperation({ summary: 'Streamable download of a preview file.' })
+	@ApiResponse({ status: 200, type: StreamableFile })
+	@ApiResponse({ status: 206, type: StreamableFile })
+	@ApiResponse({ status: 400, type: ApiValidationError })
+	@ApiResponse({ status: 403, type: ForbiddenException })
+	@ApiResponse({ status: 404, type: NotFoundException })
+	@ApiResponse({ status: 406, type: NotAcceptableException })
+	@ApiResponse({ status: 500, type: InternalServerErrorException })
+	@Get('/preview/:fileRecordId/:fileName')
+	async downloadPreview(
+		@Param() params: DownloadFileParams,
+		@CurrentUser() currentUser: ICurrentUser,
+		@Query() previewParams: PreviewParams,
+		@Req() req: Request,
+		@Res({ passthrough: true }) response: Response
+	): Promise<StreamableFile> {
+		// Get Range HTTP header value to check if caller
+		// requested either partial or full data stream.
+		const bytesRange = req.header('Range');
+
+		// Call download method with either defined or undefined bytes range.
+		const fileResponse = await this.filesStorageUC.downloadPreview(
+			currentUser.userId,
+			params,
+			previewParams,
+			bytesRange
+		);
+
+		// Destroy the stream after it has been closed.
```

Comment:
Also this comment and most of the other ones (for some I'm not sure) seem unnecessary. 

I just read and really like this blog post: https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/ Here mainly point 1 is important.

I especially like this sentence from the conclusion: "Code Tells You How, Comments Tell You Why."